subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -214,14 +204,15 @@ public void receiveData(String data) {
       System.out.println(
           java.time.LocalTime.now() + "receiveDataLen:" + data.length() + "receiveData:" + data);
     }
-
-    Runnable runnable =
-        () -> {
-          parseData(data);
-        };
-    Thread parseThread = new Thread(runnable);
-
-    parseThread.start();
+    if (data != null) {
+      Runnable runnable =
+          () -> {
+            parseData(data);
+          };
+      Thread parseThread = new Thread(runnable);
```

Comment:
can be replaced with:

Thread parseThread = new Thread(() -> parseData(data));

If that doesn't work for you, let me know and I'll figure out the exact example. I can't find a useful link to it right now, but it's using a "Functional interface" (Runnable) combined with a lambda function (() ->).

Reply:
fixed as suggested